src/bench/duplicate_inputs.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2011-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <chainparams.h>
+#include <coins.h>
+#include <consensus/merkle.h>
+#include <consensus/validation.h>
+#include <miner.h>
+#include <policy/policy.h>
+#include <pow.h>
+#include <scheduler.h>
+#include <txdb.h>
+#include <txmempool.h>
+#include <util/time.h>
+#include <validation.h>
+#include <validationinterface.h>
+
+#include <boost/thread.hpp>
+
+#include <list>
+#include <vector>
+
+
+static void DuplicateInputs(benchmark::State& state)
+{
+    const CScript SCRIPT_PUB{CScript(OP_TRUE)};
+
+    // Switch to regtest so we can mine faster
+    // Also segwit is active, so we can include witness transactions
+    SelectParams(CBaseChainParams::REGTEST);
+
+    InitScriptExecutionCache();
+
+    boost::thread_group thread_group;
+    CScheduler scheduler;
+    const CChainParams& chainparams = Params();
+    {
+        ::pblocktree.reset(new CBlockTreeDB(1 << 20, true));
+        ::pcoinsdbview.reset(new CCoinsViewDB(1 << 23, true));
+        ::pcoinsTip.reset(new CCoinsViewCache(pcoinsdbview.get()));
+
+        thread_group.create_thread(boost::bind(&CScheduler::serviceQueue, &scheduler));
```

Comment:
µnit: Could probably use something like `[&] { scheduler.serviceQueue(); }` to avoid `boost::bind`. Or if you prefer bind, use `std::bind`.

```suggestion
        thread_group.create_thread([&] { scheduler.serviceQueue(); });
```